larpandoradlcontent/LArVertex/DlVertexingAlgorithm.cc

```diff
@@ -0,0 +1,932 @@
+/**
+ *  @file   larpandoradlcontent/LArVertex/DlVertexingAlgorithm.cc
+ *
+ *  @brief  Implementation of the deep learning vertexing algorithm.
+ *
+ *  $Log: $
+ */
+
+#include <cmath>
+
+#include <torch/script.h>
+#include <torch/torch.h>
+
+#include "larpandoracontent/LArHelpers/LArFileHelper.h"
+#include "larpandoracontent/LArHelpers/LArGeometryHelper.h"
+#include "larpandoracontent/LArHelpers/LArMvaHelper.h"
+#include "larpandoracontent/LArHelpers/LArVertexHelper.h"
+
+#include "larpandoradlcontent/LArVertex/DlVertexingAlgorithm.h"
+
+using namespace pandora;
+using namespace lar_content;
+
+namespace lar_dl_content
+{
+
+DlVertexingAlgorithm::DlVertexingAlgorithm():
+    m_trainingMode{false},
+    m_trainingOutputFile{""},
+    m_event{-1},
+    m_pass{1},
+    m_height{256},
+    m_width{256},
+    m_maxHitAdc{500.f},
+    m_regionSize{32.f},
+    m_visualise{false},
+    m_writeTree{false},
+    m_rng(static_cast<std::mt19937::result_type>(0))
```

Comment:
Does the (0) here mean the seed is always 0?

Reply:
It does

Reply:
It ensures training is reproducible (whether this is genuinely important... maybe?)